client/src/util/data_layer_util.ts

```diff
@@ -1,20 +1,46 @@
+import { FeaturePropertyDataType, LegendBucketData, LegendInfo } from '@/model/data_layer';
+
 /**
- * Converts list of legend pairs (alternating numerical key, string hex color
- * value) to a map of legend key : color hex.
- * @param colorMapping the list of legend pairs
- *
- * Example:
- * [0, '#9fcd7c', 10000, '#f7e5af'] => {'0' :  '#9fcd7c', '10000', '#f7e5af'}
+ * Adds a bucketLabel to a legends bucket based on its data type.
+ * @param legend: has information on the data type for the buckets.
  */
-export function colorMapToBuckets(colorMapping: any[]): Map<string, string> {
-  const listOfLegendPairs: [string, string][] = [];
-  for (let index = 0; index < colorMapping.length; index++) {
-    if (index % 2 != 0) {
-      listOfLegendPairs.push([colorMapping[index - 1].toString(), colorMapping[index]]);
+export const formatLegendBucket = (legend: LegendInfo | undefined): Array<LegendBucketData> => {
+  legend?.buckets?.forEach((bucket) => {
+    switch (legend.bucketLabelType) {
+      case FeaturePropertyDataType.String: {
+        bucket.bucketLabel = bucket.bucketValue ?? '';
+        break;
+      }
+      case FeaturePropertyDataType.Number: {
+        bucket.bucketLabel = parseFloat(bucket.bucketValue ?? '0').toLocaleString();
+        break;
+      }
+      case FeaturePropertyDataType.Percent: {
```

Comment:
what field comes to FE as % already?

Reply:
It's not from the DB it's from the interpolation https://github.com/BlueConduit/open-data-platform/pull/64/files#diff-8e016a7a6487cb36ceffea3535fbc4d7e4a70f3b73419ff3dc2e8b768472b087

Reply:
ohh nice - I added logic to do *100 for percentages in the handler to parse the property values - how do you feel about that vs this? I might be misunderstanding, but does modifying that value in order to color the map also change the value in the feature property list?